src/memcache.c

```diff
@@ -1233,11 +1260,11 @@ static mmc_t *php_mmc_pool_addserver(
 		return NULL;
 	}
 	/* initialize pool if need be */
-	if ((connection = zend_hash_str_find(Z_OBJPROP_P(mmc_object), "connection", sizeof("connection")-1)) == NULL) {
+	if (Z_TYPE_P(connection = Z_MEMCACHE_CONNECTION_P(mmc_object)) != IS_RESOURCE) {
 		pool = mmc_pool_new();
 		pool->failure_callback = (mmc_failure_callback) &php_mmc_failure_callback;
 		list_res = zend_register_resource(pool, le_memcache_pool);
-		add_property_resource(mmc_object, "connection", list_res);
+		ZVAL_RES(Z_MEMCACHE_CONNECTION_P(mmc_object), list_res);
```

Comment:
original code would free the old value of $memcache->connection, this replacement just overwrites without checking the old value. (e.g. `$memcache->connection = new stdClass(); $memcache->addServer(...);` would no longer free the stdClass)

It isn't performance sensitive for something that'd be called once total per Memcache instance